modules/context/repo.go

```diff
@@ -282,11 +282,8 @@ func RetrieveTemplateRepo(ctx *Context, repo *models.Repository) {
 }
 
 // ComposeGoGetImport returns go-get-import meta content.
-func ComposeGoGetImport(owner, repo string) string {
-	/// setting.AppUrl is guaranteed to be parse as url
-	appURL, _ := url.Parse(setting.AppURL)
-
-	return path.Join(appURL.Host, setting.AppSubURL, url.PathEscape(owner), url.PathEscape(repo))
+func ComposeGoGetImport(ctx *Context, owner, repo string) string {
+	return path.Join(ctx.Req.Host, url.PathEscape(owner), url.PathEscape(repo))
```

Comment:
this wont work for suburls etc ...

-> Related Integration Test also fail!

Reply:
> this wont work for gitea installations using suburl etc ...

I see. Maybe appending setting.AppSubURL again will work.  But when using suburls the import path must then contain the suburl for `go get` to work.  I'll append the AppSubURL and let's look at the result of the integration tests.

But apart from that I think I realize that an external redirector is the way to go and 
gitea works fine if he module import path matches the repository path ...

Or do you have a suggestion how this can be implemented ?